code/modules/mob/transform_procs.dm

```diff
@@ -26,24 +26,25 @@
 	dropBorers()
 	return ..()
 
-/mob/proc/Postmorph(var/mob/new_mob = null)
+/mob/proc/Postmorph(var/mob/new_mob = null, var/namepick = FALSE)
 	if(!new_mob)
 		return
 	if(mind)
 		mind.transfer_to(new_mob)
 		//namepick
-		spawn(10)
-			var/newname
-			for(var/i = 1 to 3)
-				newname = reject_bad_name(stripped_input(new_mob,"You have been turned into a monkey! Pick a monkey name for your new monkey self.", "Name change [4-i] [0-i != 1 ? "tries":"try"] left",""),1,MAX_NAME_LEN)
-				if(!newname || newname == "")
-					if(alert(new_mob,"Are you sure you want to keep the default name?",,"Yes","No") == "Yes")
-						break
-				else
-					if(alert(new_mob,"Do you really want the name:\n[newname]?",,"Yes","No") == "Yes")
-						break
-			if(newname)
-				new_mob.name = new_mob.real_name = newname
+		if(namepick)
+			spawn(10)
+				var/newname
+				for(var/i = 1 to 3)
+					newname = reject_bad_name(stripped_input(new_mob,"You have been turned into a monkey! Pick a monkey name for your new monkey self.", "Name change [4-i] [0-i != 1 ? "tries":"try"] left",""),1,MAX_NAME_LEN)
```

Comment:
Might want to change the message here if this is used for things other than monkeys.

Reply:
fair